metricbeat/module/envoyproxy/server/server_integration_test.go

```diff
@@ -29,27 +29,32 @@ import (
 	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
 )
 
-func TestData(t *testing.T) {
+func TestFetch(t *testing.T) {
 	compose.EnsureUp(t, "envoyproxy")
 
-	f := mbtest.NewEventFetcher(t, getConfig())
-	err := mbtest.WriteEvent(f, t)
-	if err != nil {
-		t.Fatal("write", err)
+	f := mbtest.NewReportingMetricSetV2(t, getConfig())
+	events, errs := mbtest.ReportingFetchV2(f)
+	if len(errs) > 0 {
+		t.Fatalf("Expected 0 error, had %d. %v\n", len(errs), errs)
 	}
+	assert.NotEmpty(t, events)
+
+	t.Logf("%s/%s event: %+v", f.Module().Name(), f.Name(), events[0])
 }
 
-func TestFetch(t *testing.T) {
+func TestData(t *testing.T) {
```

Comment:
As commented in your other PR's, now `TestData` must look like this (sorry for the misleading information at the beginning):

```go
func TestData(t *testing.T) {
	f := mbtest.NewReportingMetricSetV2Error(t, getConfig())

	if err := mbtest.WriteEventsReporterV2Error(f, t, ""); err != nil {
		t.Fatal("write", err)
	}
}
```

Reply:
The above code is for the new error interface and will not work here.

Reply:
Updated, in any case, better to merge it like this and do a follow up PR.